src/Map.jsx

```diff
@@ -74,7 +82,40 @@ class Map extends Component {
       { force: true },
     );
   }
+  getStopInfo(route, info) {
+    let sInfo = {};
+    if (!this.state.stopInfo.isFirstStpChosen()) {
+      sInfo = update(this.state.stopInfo, { firstStop: { $set: info } });
+      const station = this.findStationID(route, info);
+      console.log(`First Stop ID: ${station.sid}`);
+    } else if (this.isSameStop(info, this.state.stopInfo.firstStop)) {
+      sInfo = update(this.state.stopInfo, { firstStop: { $set: {} } });
```

Comment:
So I assume you used this to avoid calling `setState`? The thing is that selecting a stop will cause some change in the map, so I guess it's okay to trigger a re-render when selecting a stop?

Reply:
@EddyIonescu It's not to avoid rerender. I call setState later.
It's because state objects are immutable in react. Every time you want to update one attribute of an object in the state of a react component, you need to copy over the entire object. You can't just edit the one attribute. Update simplifies the syntax of this work by letting you reference the object in addition to the attribute you wish to update/change